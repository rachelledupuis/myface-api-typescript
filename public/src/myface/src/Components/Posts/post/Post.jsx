import React from 'react'

export function Post( {post} ) {
    return (
        <div>
            <img src={post.imageUrl} />
            <p>posted {post.createdAt}</p>
            <p>{post.message}</p>
        </div>
    )
}