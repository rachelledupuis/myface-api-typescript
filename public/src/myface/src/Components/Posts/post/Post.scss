@import './../../constants.scss';

.post {
    display: grid;
    background-color: $darkbackground;
    grid-template-columns: 1fr; 
    grid-template-rows: 50% 1fr 2fr;  
    width: 100%;
    margin-bottom: 24px;
}

.image {
    grid-template-areas: 1 / 1 / 2 / 2;
    width: 100%;
    height: 100%;
}

.user-info{
    grid-template-areas: 2 / 1 / 3 / 2;
}

.message {
    grid-template-areas: 3 / 1 / 4 / 2;
}