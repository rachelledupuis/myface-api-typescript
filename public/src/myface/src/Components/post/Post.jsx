import React from 'react'

export function Post( {posts} ) {
    return (
        <div>
            <h1>{posts.message}</h1>
            <img src={posts.imageUrl} />
            <p>posted by {posts.postedBy.name} at {posts.createdAt}</p>
        </div>
    )
}