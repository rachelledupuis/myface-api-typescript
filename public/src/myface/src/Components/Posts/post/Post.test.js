import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Post } from "./Post";

test("renders message", () => {
  // Arrange
  const post = {
    message: "Post message",
    imageUrl: "Image url",
    createdAt: "2022/04/05 11:00",
    postedBy: {
      name: "User name",
    },
  };

  // Act
  render(<Post post={post} />);
  const messageElement = screen.getByText(/post message/i);

  // Assert
  expect(messageElement).toBeInTheDocument();
});


test("matches snapshot", () => {
  // Arrange
  const post = {
    message: "Post message",
    imageUrl: "Image url",
    createdAt: "2022/04/05 11:00",
    postedBy: {
      name: "User name",
    },
  };

  // Act
  const tree = renderer
    .create(<Post post={post} />)
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});