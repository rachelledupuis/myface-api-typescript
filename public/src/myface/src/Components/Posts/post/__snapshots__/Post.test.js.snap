// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="post"
>
  <img
    className="image"
    src="Image url"
  />
  <p
    className="user-info"
  >
    Posted by 
    User name
     on 
    2022/04/05 11:00
  </p>
  <p
    className="message"
  >
    Post message
  </p>
</div>
`;
